From f8666234251c56f89ed64580940e829a13caf521 Mon Sep 17 00:00:00 2001
From: Florian Westphal <fwestpha@redhat.com>
Date: Tue, 2 Sep 2014 10:15:08 -0400
Subject: [net] gre: When GRE csum is present count as encap layer wrt csum

Message-id: <1409652920-431-46-git-send-email-fwestpha@redhat.com>
Patchwork-id: 90157
O-Subject: [PATCH 45/57] [RHEL7.1 net] gre: When GRE csum is present count as encap layer wrt csum
Bugzilla: 1131999
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>

BZ: 1131999

commit 48a5fc773190bd5339869003fa65d38559bb8890
Author: Tom Herbert <therbert@google.com>
Date:   Fri Aug 22 13:34:52 2014 -0700

    gre: When GRE csum is present count as encap layer wrt csum

    In GRE demux if the GRE checksum pop rcv encapsulation so that any
    encapsulated checksums are treated as tunnel checksums.

    Signed-off-by: Tom Herbert <therbert@google.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv4/gre_demux.c b/net/ipv4/gre_demux.c
index e25b80e..2ef87d7 100644
--- a/net/ipv4/gre_demux.c
+++ b/net/ipv4/gre_demux.c
@@ -124,6 +124,7 @@ static int parse_gre_header(struct sk_buff *skb, struct tnl_ptk_info *tpi,
    *csum_err = true;
    return -EINVAL;
   }
+  skb_pop_rcv_encapsulation(skb);
   options++;
  }
 
-- 
1.7.1